cumulus_library/actions/builder.py

```diff
@@ -266,25 +375,49 @@ def _run_raw_queries(
             f"`{manifest.get_study_prefix()}__",
             "`",
         )
-    # We'll explicitly create a cursor since recreating cursors for each
-    # table in a study is slightly slower for some databases
-    cursor = config.db.cursor()
-    # We want to only show a progress bar if we are :not: printing SQL lines
-    with base_utils.get_progress_bar(disable=config.verbose) as progress:
-        task = progress.add_task(
-            f"Building tables from {filename}...",
-            total=len(queries),
-            visible=not config.verbose,
+    if prepare:
+        _render_output(
+            manifest.get_study_prefix(), [q[0] for q in queries], data_path, filename, query_count
         )
-        _execute_build_queries(
-            config=config,
-            manifest=manifest,
-            cursor=cursor,
-            queries=queries,
-            progress=progress,
-            task=task,
+    else:
+        # We'll explicitly create a cursor since recreating cursors for each
+        # table in a study is slightly slower for some databases
+        cursor = config.db.cursor()
+        # We want to only show a progress bar if we are :not: printing SQL lines
+        with base_utils.get_progress_bar(disable=config.verbose) as progress:
+            task = progress.add_task(
+                f"Building tables from {filename}...",
+                total=len(queries),
+                visible=not config.verbose,
+            )
+            _execute_build_queries(
+                config=config,
+                manifest=manifest,
+                cursor=cursor,
+                queries=queries,
+                progress=progress,
+                task=task,
+            )
+    return len(queries)
+
+
+def _render_output(study_name, outputs, data_path, filename, count, is_toml=False):
+    if is_toml:
+        suffix = "toml"
+    else:
+        suffix = "sql"
+    for output in enumerate(outputs):
+        if is_toml:
+            name = "config"
+        else:
+            name = re.search(r"(.*)__\w*", output[1])[0].lower().replace(" ", "_")
+        file_path = data_path / (
+            f"{study_name}/{count+output[0]:04d}.{filename.rsplit('.', 1)[0]}"
+            f".{output[0]:02d}.{name}.{suffix}"
         )
```

Comment:
These lines are pretty dense - I think the regex search line could use a small comment about like "# search for table name" or something. And I think the file path building might benefit from intermediate variables, like:
```
total_index = count + index
new_filename = f"{total_index:04d}.{filename.stem}.{index:02d}.{name}.{suffix}" 
file_path = data_path / study_name / new_filename
```